### PR TITLE
Add ingress overlay for results.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

This should allow results to have its own DNS name + TLS for serving
results externally.

Largely based off of https://github.com/tektoncd/plumbing/blob/main/tekton/cd/dashboard/overlays/dogfooding/ingress.yaml

I'm not 100% confident that this is correct, so would definitely appreciate a thorough look at this PR. 🙏 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [-] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/cd/results/overlays/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  tls:
+  - secretName: tekton-results-tls
+    hosts:
+    - results.dogfooding.tekton.dev
+  rules:
+  - host: results.dogfooding.tekton.dev
+    http:
+      paths:
+      - backend:
+          serviceName: tekton-results-api
```
Comment (nikhil-thomas) [inline tekton/cd/results/overlays/ingress.yaml:21]: it is `tekton-results-api-service`, isn't it. 🧑‍💻

Comment (wlynch) [reply tekton/cd/results/overlays/ingress.yaml]: Good catch! Fixed.

**Comment by afrittoli** [comment]
Sorry, I was late to review this one.

The ingress looks good, but the overlay should have one folder per environment, like pipeline and other projects:
```
- base
- overlays
      |- dogfooding
      \- robocat
```

It's fine to have only `dogfooding` if that's the only place we deploy to for now.
Overlays are deployed along with a release through the `deploy-tekton-release` trigger template, which is executed:
- on demand, using the [`deploy-release.sh`](https://github.com/tektoncd/plumbing/blob/main/scripts/deploy-release.sh) script, on the `dogfooding` cluster
- nightly, via the `tekton-release-cd-trigger-robotcat-*` cronjobs, on the `robocat` cluster

If you only want the ingress to be synced to the cluster, you need to place it somewhere it will be picked by the hourly job that deploys everything under `tekton/resources`, as specified by the chain of `kustomization.yaml` files.

**Pull request merged**